File: oss/src/main/java/com/truenine/component/oss/amazon/S3StatementArgs.java
package com.truenine.component.oss.amazon;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.gson.annotations.SerializedName;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class S3StatementArgs {

  @JsonProperty("Effect")
  @SerializedName("Effect")
  private String effect;

  @JsonProperty("Action")
  @SerializedName("Action")
  private List<String> action = new ArrayList<>();

  @JsonProperty("Principal")
  @SerializedName("Principal")
  private S3PrincipalArgs principal;

  @JsonProperty("Resource")
  @SerializedName("Resource")
  private List<String> resource = new ArrayList<>();
}

File: oss/shared/src/main/kotlin/io/github/truenine/composeserver/oss/S3StatementArgs.kt
package io.github.truenine.composeserver.oss

import com.fasterxml.jackson.annotation.JsonProperty

data class S3StatementArgs(
  @JsonProperty("Effect") var effect: String? = null,
  @JsonProperty("Action") var action: MutableList<String> = ArrayList(),
  @JsonProperty("Principal") var principal: S3PrincipalArgs? = null,
  @JsonProperty("Resource") var resource: MutableList<String> = ArrayList(),
)
